### PR TITLE
feat: Support Component Lock with `metadata.locked`

### PR DESCRIPTION
**Pull request by milldr**
## what
- Added support for `metadata.locked` with components

```yaml
# Lock a production database component to prevent accidental changes
components:
  terraform:
    rds:
      metadata:
        type: real
        locked: true
      vars:
        name: production-database
```

## why
- The `metadata.locked` parameter prevents changes to a component while still allowing read operations. When a component is locked, operations that would modify infrastructure (like `terraform apply`) are blocked, while read-only operations (like `terraform plan`) remain available. By default, components are unlocked. Setting `metadata.locked` to `true` prevents any change operations.

## references
- DEV-2849

### CODE REVIEW

File: examples/demo-metadata/stacks/deploy/prod.yaml
Comment (osterman) [inline examples/demo-metadata/stacks/deploy/prod.yaml:1]: Copying an entire example from something else to only add metadata as an example is too much technical debt to maintain, and doesn't focus the example enough on one small feature, which makes it difficult to understand what's necessary. In other words, all the other stuff like variables and outputs, etc. detract from the example. 

This one feature should just be added to example/tests. Alternatively, a reduced example specifically focused on locking or unlocking could be created. But a generic example on metadata is far too broad.

Comment (milldr) [reply examples/demo-metadata/stacks/deploy/prod.yaml:1]: I agree, but I was thinking this would be a good place to start adding examples for metadata usage other than just `metadata.locked`, since there are many use cases. 

Although, simple tests in example/tests would be fine by me too. Let me know which

Comment (milldr) [reply examples/demo-metadata/stacks/deploy/prod.yaml:1]: Tests failed anyway. Fixing those now

Comment (milldr) [reply examples/demo-metadata/stacks/deploy/prod.yaml:1]: Alternatively, we also could have some example where we can define many advantaged usages, including metadata and relatives paths. I dont see any of the existing examples aligning well. We shouldn't over complicate `examples/quick-start-advanced`, and `examples/tests` intentionally contains invalid configuration.

Instead we could add 1 example, such as `examples/test-advanced` where it's a valid configuration with many advanced usage validation

Comment (osterman) [reply examples/demo-metadata/stacks/deploy/prod.yaml:1]: > examples/tests intentionally contains invalid configuration.

The `examples/tests` directory is intentionally designed to contain invalid configurations. This is because tests are meant to validate both functional and non-functional scenarios, including edge cases and failure states. Adding configurations there—even if they are invalid—is exactly how we ensure robust testing of different scenarios.

In contrast, providing demos or examples serves a different purpose. Demos are optimized to showcase minimal, functional use cases for specific features, serving as a reference or starting point for users. Their goal is not to validate edge cases or invalid inputs.

The label `test-advanced` feels arbitrary to me, especially since there are already many advanced examples under `examples/tests`. The term "advanced" is highly subjective, and we aim for clarity and alignment with the intended use case. For instance, we already differentiate `quick-start-simple` and `quick-start-advanced` under the broader "quick-start" use case. 

Also, I forgot to mention we need to update the JSON Schema.

https://github.com/cloudposse/atmos/blob/main/website/static/schemas/atmos/atmos-manifest/1.0/atmos-manifest.json

Comment (milldr) [reply examples/demo-metadata/stacks/deploy/prod.yaml:1]: My confusion comes from trying to create tests with the new `tests/test_cases.yaml` functionality. 

Those tests simply run an atmos command and then check for expected output, but in order to run _any_ atmos command, atmos first must compile YAML for all stacks. Therefore `examples/tests` cannot be used for `tests/test_cases.yaml` since any command will always fail to compile intentionally.

Instead we can create or reuse a different example (what I was suggesting) or only include `go` tests (do not add `tests/test_cases`)

File: tests/fixtures/scenarios/metadata/atmos.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - "**/_defaults.yaml"
+  name_pattern: "{stage}"
+
+logs:
+  file: "/dev/stderr"
+  level: Info
+
+schemas:
+  atmos:
+    manifest: "../../schemas/atmos/atmos-manifest/1.0/atmos-manifest.json"
```
Comment (osterman) [inline tests/fixtures/scenarios/metadata/atmos.yaml:25]: Atmos will auto load the schema now. Unless we are specifically testing schema validation, let’s remove this as superfluous.

Comment (milldr) [reply tests/fixtures/scenarios/metadata/atmos.yaml:25]: Is that true? I'm not sure it's working without it yet:
```
atmos validate stacks
Schema file 'stacks/schemas/atmos/atmos-manifest/1.0/atmos-manifest.json' not found. Configure via:
1. 'schemas.atmos.manifest' in atmos.yaml
2. ATMOS_SCHEMAS_ATMOS_MANIFEST env var
3. --schemas-atmos-manifest flag

Accepts: absolute path, paths relative to base_path, or URL
```

File: internal/exec/helmfile.go [hunk 2]
```diff
@@ -94,6 +94,16 @@ func ExecuteHelmfile(info schema.ConfigAndStacksInfo) error {
 			"by 'metadata.type: abstract' attribute", filepath.Join(info.ComponentFolderPrefix, info.Component))
 	}
 
+	// Check if the component is locked (`metadata.locked` is set to true)
+	if info.ComponentIsLocked {
+		// Allow read-only commands, block modification commands
+		switch info.SubCommand {
+		case "sync", "apply", "deploy", "delete", "destroy", "secrets":
```
[the remark this answers is not in the document]
Comment (milldr) [reply internal/exec/helmfile.go:101]: Let's remove secrets, since it's not a core command

https://helmfile.readthedocs.io/en/latest/#secrets

File: tests/fixtures/scenarios/metadata/atmos.yaml (hunk 1 above)
Comment (milldr) [reply tests/fixtures/scenarios/metadata/atmos.yaml:25]: where is the remote schema defined? I cant get that to work without specifying it with one of the given options

Comment (osterman) [reply tests/fixtures/scenarios/metadata/atmos.yaml:25]: Ah, so because this PR is not yet merged, the remote schema will fail.

We really need to change the remote schema to a local `go:embed`

Comment (milldr) [reply tests/fixtures/scenarios/metadata/atmos.yaml:25]: what should we do here? Update remote schema, remove and admin merge without checks passing, or ignore for now?

Comment (osterman) [reply tests/fixtures/scenarios/metadata/atmos.yaml:25]: I'd prefer that we fix the schema handling. @aknysh can you implement the `go:embed` this week?
